src/setup/init_fluid.f90

```diff
@@ -1,468 +1,514 @@
+        real(rt),         intent(in   ) :: dx, dy, dz, xlo, ylo, zlo
+        real(rt),         intent(inout) ::    vel(slo(1):shi(1), slo(2):shi(2),slo(3):shi(3), 3)
+        real(rt),         intent(inout) :: tracer(slo(1):shi(1), slo(2):shi(2),slo(3):shi(3), ntrac)
 
-      ! Local variables
-      integer(c_int)                  :: i, j, k
-      integer(c_int)                  :: num_cells, num_cells_x, num_cells_y,num_cells_z
-      real(rt)                        :: x, y, z, yval, zval, pi
+        ! Local variables
+        integer(c_int)                  :: i, j, k
+        integer(c_int)                  :: num_cells, num_cells_x, num_cells_y,num_cells_z
+        real(rt)                        :: x, y, z, yval, zval, pi
 
-      pi = 4.0_rt * atan(one)
+        pi = 4.0_rt * atan(one)
 
-      do k = lo(3), hi(3)
-         z = zlo + dz*(real(k,rt) + 0.5d0) 
-         do j = lo(2), hi(2)
-            y = ylo + dy*(real(j,rt) + 0.5d0)
-            do i = lo(1), hi(1)
-               x = xlo + dx*(real(i,rt) + 0.5d0)
-
-               vel(i,j,k,1) = 8.0d0*(5.0d0*(1.0d0-z**4)/4.0d0 + 0.3d0*cos(5.0d0*pi*y)*exp(0.5-32.4*(1.0d0-abs(z))**2)*(1.0d0-abs(z)))
-               vel(i,j,k,2) = 8.0d0*21.6d0*sin(4.0d0*pi*x)*exp(-32.4d0*(1.0d0-abs(z))**2)*(1.0d0-abs(z))
-               vel(i,j,k,3) = 0.0d0
-               tracer(i,j,k,1:ntrac) = 0.0d0
-               if (ntrac .gt. 0 .and. i .le. domhi(1)/4) &
-                  tracer(i,j,k,1) = 1.d0
-               if (ntrac .gt. 1 .and. i .le. domhi(1)/2) &
-                  tracer(i,j,k,2) = 2.d0
-               if (ntrac .gt. 2 .and. i .le. 3*domhi(1)/4) &
-                  tracer(i,j,k,3) = 3.d0
+        do k = lo(3), hi(3)
+            z = zlo + dz*(real(k,rt) + 0.5d0)
+            do j = lo(2), hi(2)
+                y = ylo + dy*(real(j,rt) + 0.5d0)
+                do i = lo(1), hi(1)
+                    x = xlo + dx*(real(i,rt) + 0.5d0)
+
+                    vel(i,j,k,1) = 8.0d0*(5.0d0*(1.0d0-z**4)/4.0d0 + 0.3d0*cos(5.0d0*pi*y)*exp(0.5-32.4*(1.0d0-abs(z))**2)*(1.0d0-abs(z)))
+                    vel(i,j,k,2) = 8.0d0*21.6d0*sin(4.0d0*pi*x)*exp(-32.4d0*(1.0d0-abs(z))**2)*(1.0d0-abs(z))
+                    vel(i,j,k,3) = 0.0d0
+                    tracer(i,j,k,1:ntrac) = 0.0d0
+                    if (ntrac .gt. 0 .and. i .le. domhi(1)/4) &
+                        tracer(i,j,k,1) = 1.d0
+                    if (ntrac .gt. 1 .and. i .le. domhi(1)/2) &
+                        tracer(i,j,k,2) = 2.d0
+                    if (ntrac .gt. 2 .and. i .le. 3*domhi(1)/4) &
+                        tracer(i,j,k,3) = 3.d0
+                end do
+            end do
+        end do
+
+    end subroutine channel_flow
+
+    ! Initialize potential temperature with a perturbation factor
+    ! RBC domain should have height of unity and be solved with non-dimensional variables
+    subroutine rayleigh_benard(lo, hi, vel, tracer, slo, shi, dx, dy, dz, xLo, yLo, zLo, domlo)
```

Comment:
Eclipse must have some fortran format modification that I haven't changed yet.  This routine is the only real change in this file.